src/Simplify_Select.cpp

```diff
@@ -191,6 +210,7 @@ Expr Simplify::visit(const Select *op, ExprInfo *bounds) {
              (no_overflow_int(op->type) &&
               (rewrite(select(x, y * c0, c1), select(x, y, fold(c1 / c0)) * c0, c1 % c0 == 0) ||
                rewrite(select(x, c0, y * c1), select(x, fold(c0 / c1), y) * c1, c0 % c1 == 0) ||
+               rewrite(select(x, y + c0, c1), select(x, y, fold(c1 - c0)) + c0) ||
```

Comment:
This rule is causing the correctness_simplify failures. We should either remove it or possibly add a `c1 != 0` predicate.